stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveData.kt

```diff
@@ -238,7 +238,41 @@ internal class MessageListItemLiveData(
             }
         }
 
-        return messagesCopy
+        return addMessageReadFlags(messagesCopy, reads)
```

Comment:
I don't like that our logic becomes more and more complicated, loading UI thread. Seems right now we don't have any other choice. But in the future coroutine usage (with `Flow` for example) can help us with unloading UI thread.

Reply:
Yep, but we'll have to change things a bit to avoid concurrency issues